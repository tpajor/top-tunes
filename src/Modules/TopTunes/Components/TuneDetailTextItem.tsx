import * as React from 'react';

interface IProps {
  title: string;
  text: string | number;
}

const TuneDetailTextItem = ({ title, text }: IProps) => {
  return (
  <div className='container container--modal-item'>
    <h2 className='text text--modal-item-title'>
      {title}
    </h2>

    <p className={`text ${title === 'Producer' ? 'text--modal-item-rights' : 'text--modal-item-text'}`}>
      {text}
    </p>
  </div>
  );
};

export default TuneDetailTextItem;
