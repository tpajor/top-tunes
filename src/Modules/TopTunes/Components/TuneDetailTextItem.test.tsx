import * as React from 'react';
import { shallow } from 'enzyme';

import TuneDetailTextItem, { IProps } from './TuneDetailTextItem';

function setup() {
  const props: IProps = {
    title: 'Mock Title',
    text: 'Some Text',
  };

  const wrapper = shallow(<TuneDetailTextItem {...props} />);

  return {
    wrapper,
    props,
  };
}

describe('App component', () => {
  it('should match snapshot if no change was intended', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render different class if props title is "Producer"', () => {
    const { wrapper, props } = setup();

    wrapper.setProps({ ...props, title: 'Producer' });

    expect(wrapper.find('p').hasClass('text text--modal-item-rights')).toBe(true);
  });
});
