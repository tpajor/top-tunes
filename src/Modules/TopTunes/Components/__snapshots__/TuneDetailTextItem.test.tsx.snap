// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match snapshot if no change was intended 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TuneDetailTextItem
    text="Some Text"
    title="Mock Title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="text text--modal-item-title"
        >
          Mock Title
        </h2>,
        <p
          className="text text--modal-item-text"
        >
          Some Text
        </p>,
      ],
      "className": "container container--modal-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Mock Title",
          "className": "text text--modal-item-title",
        },
        "ref": null,
        "rendered": "Mock Title",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Some Text",
          "className": "text text--modal-item-text",
        },
        "ref": null,
        "rendered": "Some Text",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="text text--modal-item-title"
          >
            Mock Title
          </h2>,
          <p
            className="text text--modal-item-text"
          >
            Some Text
          </p>,
        ],
        "className": "container container--modal-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Mock Title",
            "className": "text text--modal-item-title",
          },
          "ref": null,
          "rendered": "Mock Title",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Some Text",
            "className": "text text--modal-item-text",
          },
          "ref": null,
          "rendered": "Some Text",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
